qiling/arch/arm.py

```diff
@@ -3,107 +3,111 @@
 # Cross Platform and Multi Architecture Advanced Binary Emulation Framework
 #
 
+from functools import cached_property
+
 from unicorn import Uc, UC_ARCH_ARM, UC_MODE_ARM, UC_MODE_THUMB, UC_MODE_BIG_ENDIAN
-from capstone import Cs, CS_ARCH_ARM, CS_MODE_ARM, CS_MODE_THUMB
-from keystone import Ks, KS_ARCH_ARM, KS_MODE_ARM, KS_MODE_THUMB
+from capstone import Cs, CS_ARCH_ARM, CS_MODE_ARM, CS_MODE_THUMB, CS_MODE_BIG_ENDIAN
+from keystone import Ks, KS_ARCH_ARM, KS_MODE_ARM, KS_MODE_THUMB, KS_MODE_BIG_ENDIAN
 
 from qiling import Qiling
-from qiling.const import QL_ARCH, QL_ENDIAN
+from qiling.const import QL_ENDIAN
 from qiling.arch.arch import QlArch
-from qiling.arch.arm_const import *
-from qiling.exception import QlErrorArch
+from qiling.arch import arm_const
+from qiling.arch.register import QlRegisterManager
 
 class QlArchARM(QlArch):
-    def __init__(self, ql: Qiling):
-        super().__init__(ql)
-
-        reg_maps = (
-            reg_map,
-        )
+    bits = 32
 
-        for reg_maper in reg_maps:
-            self.ql.reg.expand_mapping(reg_maper)
+    def __init__(self, ql: Qiling, endian: QL_ENDIAN, thumb: bool):
+        super().__init__(ql)
 
-        self.ql.reg.register_sp(reg_map["sp"])
-        self.ql.reg.register_pc(reg_map["pc"])
+        self._init_endian = endian
+        self._init_thumb = thumb
 
         self.arm_get_tls_addr = 0xFFFF0FE0
 
-    # get initialized unicorn engine
-    def get_init_uc(self) -> Uc:
-        if self.ql.archendian == QL_ENDIAN.EB:
-            mode = UC_MODE_ARM + UC_MODE_BIG_ENDIAN
-
-        elif self.ql.archtype == QL_ARCH.ARM_THUMB:
-            mode = UC_MODE_THUMB
+    @cached_property
```

Comment:
`cached_property` requires python >= 3.8. Same below.

Reply:
Indeed. As far I remember Python 3.8 is the minimal version required for Qiling to work (based on our tests suite).

Reply:
Yes, that's true. But there is still a small number of users (more than 5%) using 3.6 or 3.7 (https://pypistats.org/packages/qiling). I suppose it's the only feature that needs python>=3.8 so there is not too much work to do for compatibility with older python. [This answer](https://stackoverflow.com/questions/4037481/caching-class-attributes-in-python) may provide some insight about the compatible code.

Reply:
Thanks for the stats..! I didn't know we have these.
The stats show there is a non-negligble amount of users that still use Python 3.6 so I agree we should keep supporting it. However, as of Dec 2021 Python 3.6 has reached end-of-life [see [here](https://www.python.org/dev/peps/pep-0494/#lifespan)] and will no longer receive bugfixes or security updates.
Do we really want to keep supporting it after all...?

Reply:
Actually there is simple solution for 3.6 or 3.7. 
Plz check https://github.com/qilingframework/qiling/blob/750988e8eed42215bce5ee9531bc098002ebb2c8/qiling/arch/evm/vm/computation.py#L20-L24
I used to deal with this.

Reply:
Actually there is simple solution for 3.6 or 3.7. 
Plz check https://github.com/qilingframework/qiling/blob/750988e8eed42215bce5ee9531bc098002ebb2c8/qiling/arch/evm/vm/computation.py#L20-L24
I used to deal with this.

Reply:
Sure, threre is always a way to work around these things.
It just started a discussion about whether that would make a good opportunity to bump up the requires Python version, along with its new features.

Reply:
Yes I know, this is something worth thinking about.